pkg/cmd/docs-issue-generation/requests.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cockroachdb/errors"
+)
+
+const (
+	graphQLURL        = "https://api.github.com/graphql"
+	crlJiraBaseUrl    = "https://cockroachlabs.atlassian.net/"
+	jiraRESTURLPart   = "rest/api/3/"
+	jiraDocsUserEmail = "cockroach-jira-docs@cockroachlabs.com"
+)
+
+const (
+	httpSourceGitHub httpReqSource = iota
+	httpSourceJira
+)
+
+var (
+	tokenParams      = tokenParameters()
+	githubAuthHeader = fmt.Sprintf("Bearer %s", tokenParams.GitHubToken)
+)
+
+func tokenParameters() apiTokenParameters {
+	const (
+		githubApiTokenEnv = "GITHUB_API_TOKEN"
+		jiraApiTokenEnv   = "JIRA_API_TOKEN"
+	)
+	return apiTokenParameters{
+		GitHubToken: maybeEnv(githubApiTokenEnv, ""),
+		JiraToken:   maybeEnv(jiraApiTokenEnv, ""),
+	}
+}
+
+// queryGraphQL is the function that interfaces directly with the GitHub GraphQL API. Given a query, variables, and
+// token, it will return a struct containing the requested data or an error if one exists.
+func queryGraphQL(query string, queryVariables map[string]interface{}, out interface{}) error {
+	body := map[string]interface{}{
+		"query": query,
+	}
+	if queryVariables != nil {
+		body["variables"] = queryVariables
+	}
+	err := httpRequest(graphQLURL, "POST", httpSourceGitHub, nil, body, &out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func queryJiraRESTAPI(
+	apiEndpoint, method string, headers map[string]string, body interface{}, out interface{},
+) error {
+	url := crlJiraBaseUrl + jiraRESTURLPart + apiEndpoint
+	err := httpRequest(url, method, httpSourceJira, headers, body, &out)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func httpRequest(
+	url, method string,
+	source httpReqSource,
+	headers map[string]string,
+	body interface{},
+	out interface{},
+) error {
+	var requestBody bytes.Buffer
+	encoder := json.NewEncoder(&requestBody)
+	encoder.SetEscapeHTML(false)
+	err := encoder.Encode(body)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(method, url, &requestBody)
+	if err != nil {
+		return err
+	}
+	if source == httpSourceJira {
+		req.SetBasicAuth(jiraDocsUserEmail, tokenParams.JiraToken)
+	} else if source == httpSourceGitHub {
+		req.Header.Set("Authorization", githubAuthHeader)
+	}
```

Comment:
Please add an `else { return fmt.Errorf("unexpected httpReqSource %d", source) }` or similar to protect if someone passes in some unexpected number.